packages/backend/src/services/CoderService/CoderService.ts

```diff
@@ -0,0 +1,180 @@
+};
+
+export class CoderService extends BaseService {
+    lightdashConfig: LightdashConfig;
+
+    analytics: LightdashAnalytics;
+
+    projectModel: ProjectModel;
+
+    savedChartModel: SavedChartModel;
+
+    dashboardModel: DashboardModel;
+
+    spaceModel: SpaceModel;
+
+    schedulerClient: SchedulerClient;
+
+    constructor({
+        lightdashConfig,
+        analytics,
+        projectModel,
+        savedChartModel,
+        dashboardModel,
+        spaceModel,
+        schedulerClient,
+    }: CoderServiceArguments) {
+        super();
+        this.lightdashConfig = lightdashConfig;
+        this.analytics = analytics;
+        this.projectModel = projectModel;
+        this.savedChartModel = savedChartModel;
+        this.dashboardModel = dashboardModel;
+        this.spaceModel = spaceModel;
+        this.schedulerClient = schedulerClient;
+    }
+
+    private static transformChart(
+        chart: SavedChartDAO,
+        spaceSummary: Pick<SpaceSummary, 'uuid' | 'slug'>[],
+    ): ChartAsCode {
+        const spaceSlug = spaceSummary.find(
+            (space) => space.uuid === chart.spaceUuid,
+        )?.slug;
+        if (!spaceSlug) {
+            throw new NotFoundError(`Space ${chart.spaceUuid} not found`);
+        }
+
+        return {
+            name: chart.name,
+            description: chart.description,
+            tableName: chart.tableName,
+            updatedAt: chart.updatedAt,
+            metricQuery: chart.metricQuery,
+            chartConfig: chart.chartConfig,
+            dashboardUuid: chart.dashboardUuid,
+            colorPalette: chart.colorPalette,
+            slug: chart.slug,
+            tableConfig: chart.tableConfig,
+
+            spaceSlug,
+            version: currentVersion,
+        };
+    }
+
+    async getCharts(user: SessionUser, projectUuid: string) {
+        const project = await this.projectModel.get(projectUuid);
+        if (!project) {
+            throw new NotFoundError(`Project ${projectUuid} not found`);
+        }
+
+        // TODO handle permissions
+        // Filter charts based on user permissions (from private spaces)
+
+        // TODO
+        // We need to get the charts and all the chart config
+        // At the moment we are going to fetch them all in individual queries
+        // But in the future we should fetch them in a single query for optimiziation purposes
+        const chartSummaries = await this.savedChartModel.find({ projectUuid });
+        const chartPromises = chartSummaries.map((chart) =>
+            this.savedChartModel.get(chart.uuid),
+        );
+        const charts = await Promise.all(chartPromises);
+
+        // get all spaces to map  spaceSlug
+        const spaceUuids = charts.map((chart) => chart.spaceUuid);
+        const spaces = await this.spaceModel.find({ spaceUuids });
+        return charts.map((chart) =>
+            CoderService.transformChart(chart, spaces),
+        );
+    }
+
+    async upsertChart(
```

Comment:
Can we resuse logic from the promoting charts etc. here?

Reply:
It is 100% doable, and we should  reuse that code soon. 

I think it requires a bit of work to adapt thos emethods and fake some data, so it would look like:  

```
        let promotionChanges: PromotionChanges =
        PromoteService.getChartChanges(chartAsCode, existingChart);

    promotionChanges = await this.promoteService.upsertSpaces(user, promotionChanges);

    promotionChanges = await this.promoteService.upsertCharts(user, promotionChanges);

```

I would prioritize this over having dashboards, because then having dashboards will be really easy. And spaces will be handled automatically